src/main/scala/io/iohk/ethereum/jsonrpc/TestService.scala

```diff
@@ -161,13 +152,37 @@ class TestService(
     storeGenesisAccountStorageData(genesisData.alloc)
     // update test ledger with new config
     testLedgerWrapper.blockchainConfig = newBlockchainConfig
+    consensus.blockchainConfig = newBlockchainConfig
 
     accountAddresses = genesisData.alloc.keys.toList
     accountRangeOffset = 0
 
     SetChainParamsResponse().rightNow
   }
 
+  private def buildNewConfig(blockchainParams: BlockchainParams) = {
+    val byzantiumBlockNumber: BigInt = blockchainParams.byzantiumForkBlock.getOrElse(Int.MaxValue)
+    val istanbulForkBlockNumber: BigInt = blockchainParams.istanbulForkBlock.getOrElse(Int.MaxValue)
+
+    // For block number which are not specified by retesteth, we try to align the number to another fork
```

Comment:
What happened before this? And what should we do for future forks? What would happen if we'd also use `Int.MaxValue` here?

Reply:
Before this I think the value defined in the configuration or the value defined in the previous run. So this would work on most of tests that just use the Istanbul fork, and tests that have not depend on something specific to a previous fork (actually I am surprised so many tests pass without this fix). 

However you are right. The way it is done, if we add a new fork then in the test the number will have whatever is defined in the configuration as a default, and if it is 0 then it will activate this fork everytime (but the compiler won't catch it). I think what we can do is a small refactoring of `BlockchainConfig` to extract the block numbers in another case class `ForkBlockNumbers` and rebuild completely `ForkBlockNumbers` when updating the config in the test. This way if we ever add a new fork the compiler will catch that we should also update the TestService instead of silently copying some value.